src/Cardano/Wallet/Mnemonic.hs

```diff
@@ -0,0 +1,332 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- | Module provides mnemonic creation and
+-- | restoring from backup phrase functionality
+
+module Cardano.Wallet.Mnemonic
+       (
+       -- * Types
+         Mnemonic
+       , Entropy
+       , EntropySize
+       , MnemonicWords
+
+       -- * Errors
+       , MnemonicError(..)
+       , MnemonicException(..)
+       , EntropyError(..)
+       , DictionaryError(..)
+       , MnemonicWordsError(..)
+
+       -- * Creating @Mnemonic@ (resp. @Entropy@)
+       , mkEntropy
+       , mkMnemonic
+       , genEntropy
+
+       -- * Converting from and to @Mnemonic@ (resp. @Entropy@)
+       , mnemonicToEntropy
+       , entropyToMnemonic
+       , entropyToByteString
+       ) where
+
+
+import Prelude
+
+
+import Basement.Sized.List
+    ( unListN )
+import Control.Arrow
+    ( left )
+import Control.Monad
+    ( when, (>=>) )
+import Control.Monad.Catch
+    ( throwM )
+import Crypto.Encoding.BIP39
+import Data.Aeson
+    ( FromJSON (..), ToJSON (..) )
+import Data.Aeson.Types
+    ( Parser )
+import Data.ByteArray
+    ( constEq )
+import Data.ByteString
+    ( ByteString )
+import Data.Default
+    ( Default (def) )
+import Data.Proxy
+    ( Proxy (..) )
+import Data.Text
+    ( Text, pack, unpack )
```

Comment:
`pack` and `unpack` are ambiguous (same name in all `ByteString` packages), so let's use qualified imports for them :+1:

Reply:
done